### PR TITLE
refactor(VRV): change code structure and more

### LINKED ISSUE #4809: VRV | vrv.co
**Describe the Bug**
Doesn't display anything about the current episode you are watching on vrv.co, however it does show the info on the homepage

**Steps to Reproduce**
1. add the vrv presences 
2. go to an episode on vrv.co
3. and press play

**Expected Behavior**
should display the episode name, time left, current season, and more

**Screenshots**
my rpc while watching the episode: [link](https://i.imgur.com/3A48rOK.png)
my rpc while viewing the homepage: [link](https://i.imgur.com/SGJU5XD.png)

**Environment**
- OS: Win 11
- Browser: Brave
- Version: Version 1.31.87 Chromium: 95.0.4638.54 (Official Build) (64-bit)

**Additional context**
this only seems to be an issue while actually playing the episode, and doesn't happen on any other page of vrv

### PR DESCRIPTION
**Pull request by RisingSunLight42**
### Changelog

- Resolves #4809 by adding a iFrameRegExp in the metadata, forcing PreMid to get the right iframe for the video element
- Fix error on `presence.ts`, currentTime and duration were not correctly getted.
- Add five settings : multilanguage, Show Timestamps, Show Cover, Show Buttons, Show Search
- Add feature : display what user is searching if setting enabled
- Add feature : display cover of what the user is currently viewing if setting enabled
- Add feature : display buttons to series or episodes if setting enabled.
- Add feature : users can remove timestamps
- Refactoring the localization of the presence and adding 7 localized strings to it
- Refactoring the whole `presence.ts` by removing `if/else` in favor of `switch/case` and redoing structure to avoid duplicate code.

### Ask to change the author

@Bas950 I would like, if you're fine with that, adding me to author and you to contributor since I've edited almost everything in the presence. If you're fine with, I'll change the clientId at the top of the file and the metadata file, if not, I'll do nothing. ^^

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

Context for the first screenshot : since the search can be done on every page of the website, if the setting to show search query is disabled, the presence will display what is displayed behind the research panel.

vrv_proof_10

vrv_proof_9

vrv_proof_8

vrv_proof_7

vrv_proof_6

vrv_proof_5

vrv_proof_4

vrv_proof_3

vrv_proof_2

vrv_proof_1

</details>

### CODE REVIEW

**Comment by Bas950** [comment]
> @Bas950 I would like, if you're fine with that, adding me to author and you to contributor since I've edited almost everything in the presence. If you're fine with, I'll change the clientId at the top of the file and the metadata file, if not, I'll do nothing. ^^

Feel free to switch as around as auther and contributor.
clientId shouldn't need changing since it works... but for future maintainability maybe use imgur links for images and not the keys of the discord assets.

**Comment by RisingSunLight42** [comment]
> > @Bas950 I would like, if you're fine with that, adding me to author and you to contributor since I've edited almost everything in the presence. If you're fine with, I'll change the clientId at the top of the file and the metadata file, if not, I'll do nothing. ^^
> 
> Feel free to switch as around as auther and contributor.
> clientId shouldn't need changing since it works... but for future maintainability maybe use imgur links for images and not the keys of the discord assets.

I'll switch to imgur link then, thanks!

**Comment by RisingSunLight42** [comment]
Imgur links added and author/contributor swaped !

**Comment by EncryptedDev** [comment]
Very detailed description

File: websites/V/VRV/presence.ts [hunk 1]
```diff
@@ -1,12 +1,32 @@
...
 	}),
-	strings = presence.getStrings({
-		play: "general.playing",
-		pause: "general.paused",
-	});
+	logo = "https://i.imgur.com/p6Xv2Bv.png",
+	searchImg = "https://i.imgur.com/OIgfjTG.png",
+	playImg = "https://i.imgur.com/KNneWuF.png",
+	pauseImg = "https://i.imgur.com/BtWUfrZ.png",
+	readingImg = "https://i.imgur.com/53N4eY6.png";
```
Comment (EncryptedDev) [inline websites/V/VRV/presence.ts:8]: Put this into an array? Assets[]

Comment (RisingSunLight42) [reply websites/V/VRV/presence.ts:8]: I didn't know of that thing, thanks for the advise! I'll apply the changes as soon as possible!

**Comment by RisingSunLight42** [comment]
> Very detailed description 🚀

I prefer to detail to make the review easier for you since there's a lot of changes!

File: websites/V/VRV/presence.ts (hunk 1 above)
Comment (RisingSunLight42) [reply websites/V/VRV/presence.ts]: Resolved in https://github.com/PreMiD/Presences/pull/6643/commits/9ea0753c14014460561121db534de68384447960

**Review by Bas950 — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**